src/unix.rs

```diff
@@ -113,7 +112,7 @@ impl HttpUnixConnector {
 impl NetworkConnector for HttpUnixConnector {
     type Stream = HttpUnixStream;
 
-    fn connect(&self, host: &str, port: u16, scheme: &str) -> hyper::error::Result<HttpUnixStream> {
```

Comment:
Do you know what's going on here? It looks like these parameters were being completely ignored because this is a Unix-domain socket, and replaced with `self.path`. This is clearly a problem in the original code and I should probably fix the `NetworkConnector` API to not have a `connect` method which looks like this one at all.